compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -111,7 +111,16 @@ chooseMatmulTileParamsX86_64(EncodingUser user, TypeRange elementTypes,
     // reconsider when taking advantage of native f16/bf16 arithmetic when the
     // accumulator itself is f16/bf16.
     if (hasFeature(target, "+avx512f")) {
-      return MatmulTileParams{16, 1, 16};
+      MatmulTileParams tileParams{16, 1, 16};
+      if (!hasMicrokernels(target)) {
+        if (isVecMatEncodingUser(user)) {
```

Comment:
@bjacob, @hanhanW, I spent quite some time doing some acrobatics to get the information needed here to determine if the matmul is a vecmat/matvec and ended up with something messy that didn't work for some cases. Then, I decided to try extending the encoding to support MATVEC/VECMAT and the implementation seems simple/cleaner. Do you think this is a fair approach?

More interesting, if we go with this, we could connect `linalg.vecmat`/`linalg.matvec` to the new VECMAT/MATVEC encoding and disable the expand vector pass, which should fix the simple case of reshape ops described in https://github.com/openxla/iree/issues/15242#issuecomment-1792972109 (also @NatashaKnk, @pzread)

Thoughts? :)

Reply:
@dcaballe , I am working on a PR that makes tile size selection for vecmat / matvec / other narrow cases much more explicit and no longer implicit in the way it currently is (https://github.com/openxla/iree/blob/717c7a0dcda433b965ac14c2c3a287ae07796c8c/compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp#L272-L317). I think it will be necessary to properly solve this problem space.

Reply:
To elaborate a bit more on what I'm proposing here as the solution... We should solve this for all narrow matmuls, where narrow means "either M or N is small enough that padding is a major performance concern, particularly the case where M or N is less than tile size".  The present PR, tying this to VECMAT/MATVEC encodings, would solve this only for the cases M==1 and N==1 respectively. What about M==2, N==2, M==4, etc. To get something that scales better, I'm proposing sticking to only the MATMUL encoding user enum value, instead adding separate optional fields in EncodingAttr for static small M/N dimension sizes.  (Which can't be inferred from the shape of the tensor being encoded because they can refer to a dimension that does not exist in that tensor, for instance, the RHS tensor only has dimensions K and N, does not involve dimension M).

Reply:
https://github.com/openxla/iree/pull/15431 has my WIP stuff.

Reply:
That also sounds ok to me. Let me know when it's ready and I'll rebase on top of that. Thanks!

Reply:
@NatashaKnk @pzread (echoing here after discussion with @dcaballe ) I believe that https://github.com/openxla/iree/pull/15431 is a viable alternative to this.  Diego told me about the possibility that we may evolve SetEncoding to match matvec/vecmat named ops, not just matmul.  Doing so is neither easier nor harder after #15431.

Reply:
5x slower with DT, 3x slower with DT + UK. Let's use the issues in https://github.com/openxla/iree/issues/15313 for further discussion about this. I'll rebase this PR to use Benoit's changes.